src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java

```diff
@@ -0,0 +1,158 @@
+package org.jenkinsci.test.acceptance.recorder;
+
+import org.jenkinsci.test.acceptance.utils.SystemEnvironmentVariables;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.monte.media.Format;
+import org.monte.media.FormatKeys;
+import org.monte.media.VideoFormatKeys;
+import org.monte.media.math.Rational;
+import org.monte.screenrecorder.ScreenRecorder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.awt.GraphicsEnvironment;
+import java.awt.GraphicsConfiguration;
+import java.awt.Dimension;
+import java.awt.AWTException;
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * JUnit Rule that before executing a test it starts a recording current screen
+ * and after the test is executed, it stops recording.
+ */
+public class TestRecorderRule extends TestWatcher {
+
+    private static final Logger logger = LoggerFactory.getLogger(TestRecorderRule.class);
+
+    private static final int MAX_RECORDING_TIME_SECS = 120000;
+    private static final int FRAME_RATE_PER_SEC = 60;
+    private static final int BIT_DEPTH = 16;
+    private static final float QUALITY_RATIO = 0.97f;
+
+    private static final String TARGET = "target";
+    static final String OFF = "off";
```

Comment:
(just a comment - no need to change) personally I prefer not exposing the inner workings and then making the test use reflection to set the field - as it maintains the "this is for internal use only" contract. Others do have different opinions YMMV :-)